src/app/lib/mainsail/unconfirmed-transactions.service.ts

```diff
@@ -1,46 +1,37 @@
-import { ArkClient } from "@arkecosystem/typescript-client";
-import type { ConfigRepository } from "@/app/lib/mainsail";
+import type { ConfigRepository, Networks } from "@/app/lib/mainsail";
+
+import { ClientService } from "./client.service.js";
 import type { IProfile } from "@/app/lib/profiles/profile.contract";
 
 export class UnconfirmedTransactionsService {
-	readonly #client: ArkClient;
+	#config: ConfigRepository;
+	#profile: IProfile;
+	#client: ClientService;
 
 	constructor({ config, profile }: { config: ConfigRepository; profile: IProfile }) {
-		this.#client = new ArkClient({
-			api: config.host("tx", profile),
-			evm: config.host("evm", profile),
-		});
+		this.#config = config;
+		this.#profile = profile;
+		this.#client = new ClientService({ config: this.#config, profile: this.#profile });
 	}
 
-	async listUnconfirmed(parameters?: {
-		page?: number;
-		limit?: number;
-		offset?: number;
-		from?: string[] | string;
-		to?: string[] | string;
-	}) {
-		const normalize = (v?: string[] | string) =>
-			v == null ? undefined : (Array.isArray(v) ? v : v.split(",")).map((s) => s.trim()).filter(Boolean);
-
+	async listUnconfirmed(parameters?: { page?: number; limit?: number; from: string[]; to: string[] }) {
+		// TODO: use identifiers param instead once supported in mainsail, as it now filters based on from AND to which only works when sending to yourself
```

Comment:
this requires an adjustment in the mainsail API. Once available we can use `identifiers` instead of `from` + `to` as that causes issues